karapace/compatibility/jsonschema/checks.py

```diff
@@ -248,7 +245,6 @@ def compatibility_rec(
     # reader has type `array` to represent a list, and the writer is either a
     # different type or it is also an `array` but now it representes a tuple.
     if reader_schema is None and writer_schema is not None:
-        LOG.debug("Schema removed reader_schema.type='%r'", get_type_of(reader_schema))
```

Comment:
Is this really unnecessary? Maybe leave, but without parameters. It doesn't look like something that would fill up logs.

Reply:
- This is a recursive call, number of log lines is undetermined
- The same information is encoded in the object instantiate in the next line with additional context (namely where in the recursion is the incompatibility), and that is logged by the caller of this function